source/inmem/local.go

```diff
@@ -0,0 +1,74 @@
+package inmem
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/golang-migrate/migrate/v4/source"
+)
+
+const (
+	schemeKey = "inmem"
+	scheme    = schemeKey + "://" // construct inmem:// as scheme in Open function
+)
+
+// localMemory saves all *source.Migrations mapped by key
+type localMemory struct {
```

Comment:
About this, I doubt that it can be changed that way. Because this struct only used in this internal package, so it won't be exposed or used by the user. This struct only for `migrations` local-global variable that save migrations within a `key`. 

The usage of this is like `source.Register`, so I thought this is simplest way to add new `Migration` with specific key identifier. 

https://github.com/golang-migrate/migrate/blob/f76b94ab7deae253ea7f35c79e9df44e5195860e/source/driver.go#L96-L107